### DISCUSSION TITLE
Cant get error validation to work in child component using react-hook-form

### DISCUSSION THREAD

**Discussion by Ankitakg** [react-hook-form/react-hook-form#8320] [Q&A]
Version Number
7.30.0

Issue -

I am trying to show validation error message for the fields which are present in the child components. I am passing register and errors as a props to child components and then using the passed field name to check error. But in UI i am not able to see any error validation for empty field.Can somebody help.

Parent component -

const { register , handleSubmit , formState: { errors } } = useForm({
    mode: "onChange"
  });

 <FormFieldsComponent 
                register={register}   
                errors={ errors }
                headerName = "Final URL"
                inputName = "finalURL"
                inputPlaceholder = "https://www.example.com"
                required = "Required"
                reqMsg = "Enter a URL that's complete and correct"
                patternString = {INPUTURLPATTERN}
             />

child component --

const FormFieldsComponent = (props:any) => {

    const {register, errors , headerName ,
          inputName ,
          inputPlaceholder ,
          reqMsg,
          patternString,
          inputType,
          
             } = props

  return (
    <div className="section-container py-1 mb-2">
                <div className="form-label-row flex flex-col text-sm px-1 ml-1">
                   <div className="td-header-row flex flex-row mt-4 mb-1">
                      <label className="m-[4px] font-bold">{headerName}</label>
                      <span className="m-[4px]"><AiOutlineQuestionCircle className="relative top-[3px]" /></span>
                   </div>                           
                </div>
                <div className="content-form-row flex flex-col leading-7 ml-1">
                   <div className="h-[50px] w-[270px] p-1"> 
                    <input type={inputType} 
                      placeholder={inputPlaceholder}
                      className="w-full h-full  border border-[#80868b] rounded px-3 text-sm"
                      {...register(inputName,{
                      required: "Required",
                      pattern:{
                        value : patternString,
                        message: "Invalid"
                        }
                      })}
                      />

                     <div className="flex flex-row justify-between px-1 py-2">
                       {console.log("inputname are " + errors.inputName)}
                       {errors.inputName ? <p className="text-xs text-red-500">{errors.inputName.message}</p> : <p className="text-xs text-slate-500">Required</p>}
                     </div> 
                   </div>        
               </div>
              </div> 
  )
}

export default FormFieldsComponent;

Expected behaviour
errors should also work at child component

What browsers are you seeing the problem on?
Firefox

Relevant log output
No response

**Comment by Moshyfawn** [comment]
I'd suggest to create a reproduction codesanbox instead of a poorly formatted plain text, so there's a chance we can help you figure it out ;)
